mentat/llm_api_handler.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+import os
+import sys
+from typing import Literal, Optional, overload
+
+import sentry_sdk
+import tiktoken
+from dotenv import load_dotenv
+from openai import AsyncOpenAI, AsyncStream, AuthenticationError
+from openai.types.chat import (
+    ChatCompletion,
+    ChatCompletionChunk,
+    ChatCompletionMessageParam,
+)
+
+from mentat.errors import UserError
+from mentat.session_context import SESSION_CONTEXT
+from mentat.utils import mentat_dir_path
+
+
+def is_test_environment():
+    """Returns True if in pytest and not benchmarks"""
+    benchmarks_running = os.getenv("MENTAT_BENCHMARKS_RUNNING")
+    return (
+        "PYTEST_CURRENT_TEST" in os.environ
+        and "--benchmark" not in sys.argv
+        and (not bool(benchmarks_running) or benchmarks_running == "false")
+    )
+
+
+def raise_if_in_test_environment():
+    assert (
+        not is_test_environment()
+    ), "OpenAI call attempted in non benchmark test environment!"
+
+
+# Ensures that each chunk will have at most one newline character
+def chunk_to_lines(chunk: ChatCompletionChunk) -> list[str]:
+    content = chunk.choices[0].delta.content
+    return ("" if content is None else content).splitlines(keepends=True)
+
+
+def count_tokens(message: str, model: str, full_message: bool = False) -> int:
```

Comment:
We want the user to be able to set any model they want to though (they can set openai base to automatically change the url we request from); I think I'll leave it as a string because of this.